### PR TITLE
Check if it is same order status when updating

### LINKED ISSUE #22011: When the order status is changed in the webservice, the OrderHistory instance is duplicated
#### Describe the bug

When the order status is changed by WebService to the same one that is configured, an OrderHistory instance of the same order status is generated again:
imagen

#### Expected behavior

Do not duplicate the OrderHistory instance

#### Steps to Reproduce

As the `setCurrentState` method used for the WebService (redirected by `setWsCurrentState`) is the same one that generated this issue: https://github.com/PrestaShop/PrestaShop/issues/15513, I used the same module to reproduce this bug.

1. We installed the module that comments @khouloudbelguith https://github.com/PrestaShop/PrestaShop/issues/15513#issuecomment-534964591
2. We run the webhook: `https://prestashop.local/es/module/fgorderstate/webhook?order_id=23&token=abracadabra&method=task_status_change` changing the order id to the one we want to change the order status
3. The order will have changed status:
imagen
4. If we run the webhook again, it has created another instance of OrderHistory:
imagen

#### Additional information

* PrestaShop version: 1.7.6.9
* PHP version: 7.2.34

### PR DESCRIPTION
**Pull request by jordiweblidera**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Added check if it is the same order status you are trying to update through Webservice.
| Type?         | bug fix
| Category?     | WS
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #22011 
| How to test?  | If you send a request to the order status update web service, it does not now create an OrderHistory instance if it is the same status you want to change to

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/22006)

### CODE REVIEW

**Comment by Progi1984** [comment]
@jordiweblidera  Thank you for your contribution. :hugs: 

Could you explain in [an issue your initial bug for this PR](https://github.com/PrestaShop/PrestaShop/issues/new/choose) (or may be link to an issue) ?

Thanks :+1:

**Comment by jordiweblidera** [comment]
> 
> 
> @jordiweblidera Thank you for your contribution. 🤗
> 
> Could you explain in [an issue your initial bug for this PR](https://github.com/PrestaShop/PrestaShop/issues/new/choose) (or may be link to an issue) ?
> 
> Thanks 👍

Hi @Progi1984,

Thanks, updated

File: classes/order/Order.php [hunk 1]
```diff
@@ -1698,6 +1698,10 @@ public function setCurrentState($id_order_state, $id_employee = 0)
         if (empty($id_order_state) || (int) $id_order_state === (int) $this->current_state) {
             return false;
         }
```
Comment (Arman-Hosseini) [inline classes/order/Order.php:1700]: And the second condition must be edited.

File: classes/order/Order.php [hunk 2]
```diff
@@ -1698,6 +1698,10 @@ public function setCurrentState($id_order_state, $id_employee = 0)
...
+        $current_order_state = $this->getCurrentOrderState();
+        if ($current_order_state->id == $id_order_state) {
```
Comment (Arman-Hosseini) [inline classes/order/Order.php:1702]:
```suggestion
        if ( (int) $current_order_state->id === (int) $id_order_state) {
```

**Comment by jordiweblidera** [comment]
I'm sorry. I just saw that the bug was already corrected in `develop`: https://github.com/PrestaShop/PrestaShop/pull/20622. I close the issue and the PR.

**Pull request closed without merging**